onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc

```diff
@@ -527,12 +542,14 @@ Status GreedySearchProcessLogits(
   }
 
   cuda::LaunchLogitsProcessKernel<CudaT>(
-      reinterpret_cast<CudaT*>(next_token_scores.data()),
+      is_reuse_logits_buffer ? const_cast<CudaT*>(logits_data)
```

Comment:
maybe add a todo here for potential interface changes in ProcessLogits() in case we get a chance in future

Reply:
Need to add some comments in some other places too. Will do.